### PR TITLE
txnbuild: example docs

### PR DESCRIPTION
**Pull request by ire-and-curses**
This PR:

- adds GoDoc examples
- removes scratchpad test code
- fills in missing docs
- removes all imports of deprecated horizon/client
- updates README with example + final links (will be correct after move to main tree)

GoDocs were tested locally with `godoc -http=:6060`, `go test` and inspection in Stellar laboratory.

### CODE REVIEW

File: exp/txnbuild/README.md [hunk 1]
```diff
@@ -5,15 +5,51 @@
...
+	import "github.com/stellar/go/txnbuild"
+
+	// Make a keypair for a known account from a secret seed
+	kp, _ := keypair.Parse("SBPQUZ6G4FZNWFHKUWC5BEYWF6R52E3SEP7R3GWYSM2XTKGF5LNTWW4R")
+
+	// Get the current state of the account from the network
+	client := horizonclient.DefaultTestNetClient
+	ar := horizonclient.AccountRequest{AccountID: kp.Address()}
+	sourceAccount, err := client.AccountDetail(ar)
+	check(err)
```
Comment (poliha) [inline exp/txnbuild/README.md]: Can we just use the normal error conditional here? or a note that says what check does.
Same for other occurrence of `check(err)`

File: exp/txnbuild/create_passive_offer.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ type CreatePassiveSellOffer struct {
 	Selling Asset
 	Buying  Asset
 	Amount  string
-	Price   string // TODO: Extend to include number, and n/d fraction. See package 'amount'
+	Price   string
```
Comment (poliha) [inline exp/txnbuild/create_passive_offer.go]: Do we have an issue for adding number and fraction?

File: exp/txnbuild/example_test.go [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+		Operations:    []Operation{&op},
+		Timebounds:    NewInfiniteTimeout(), // Use a real timeout in production!
+		Network:       network.TestNetworkPassphrase,
+	}
+
+	txe, err := tx.BuildSignEncode(kp.(*keypair.Full))
+	check(err)
+	fmt.Println(txe)
+
+	// Output: AAAAAODcbeFyXKxmUWK1L6znNbKKIkPkHRJNbLktcKPqLnLFAAAAZAAMoj8AAAAEAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAJAAAAAAAAAAHqLnLFAAAAQP3NHWXvzKIHB3+jjhHITdc/tBPntWYj3SoTjpON+dxjKqU5ohFamSHeqi5ONXkhE9Uajr5sVZXjQfUcTTzsWAA=
```
Comment (poliha) [inline exp/txnbuild/example_test.go]: I think these outputs will fail when testnet is reset. We can either remove the outputs or get them from pubnet.

Comment (poliha) [reply exp/txnbuild/example_test.go]: Update. I just saw the mock horizon client. so this should be fine

File: exp/txnbuild/manage_offer.go [hunk 4]
```diff
@@ -52,7 +52,7 @@ type ManageSellOffer struct {
 	Selling Asset
 	Buying  Asset
 	Amount  string
-	Price   string // TODO: Extend to include number, and n/d fraction. See package 'amount'
+	Price   string
```
Comment (poliha) [inline exp/txnbuild/manage_offer.go]: Same as above

File: exp/txnbuild/transaction.go [hunk 5]
```diff
@@ -1,6 +1,11 @@
 /*
 Package txnbuild implements transactions and operations on the Stellar network.
-TODO: More explanation + links here
+
+This library provides an interface to the Stellar transaction model. It supports the building of Go applications on
+top of the [Stellar network](https://www.stellar.org/). Transactions constructed by this library may be submitted
+to any Horizon instance for processing onto the ledger, using any Stellar SDK client. The recommended client for Go
+programmers is [horizonclient](https://github.com/stellar/go/tree/master/clients/horizonclient). Together, these two
+libraries provide a complete Stellar SDK.
```
Comment (poliha) [inline exp/txnbuild/transaction.go]: Will godoc read these links as markdown?

File: exp/txnbuild/create_passive_offer.go (hunk 2 above)
Comment (ire-and-curses) [reply exp/txnbuild/create_passive_offer.go]: Yes - #1028

File: exp/txnbuild/example_test.go (hunk 3 above)
Comment (ire-and-curses) [reply exp/txnbuild/example_test.go]: yeah it doesn't depend on any real network

File: exp/txnbuild/manage_offer.go (hunk 4 above)
Comment (ire-and-curses) [reply exp/txnbuild/manage_offer.go]: Yes - #1028

File: exp/txnbuild/transaction.go (hunk 5 above)
Comment (ire-and-curses) [reply exp/txnbuild/transaction.go]: good catch!

**Pull request merged**